core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -16,26 +16,6118 @@
+
+        with self.swap(datetime, 'datetime', MockDatetime13Hours), self.swap(
+            db.DateTimeProperty, 'data_type', MockDatetime13Hours):
+            update_datastore_types_for_mock_datetime()
+            run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_missing_collection_model_failure(self):
+        collection_models.CollectionModel.get_by_id('0').delete(
+            feconf.SYSTEM_COMMITTER_ID, '', [])
+        expected_output = [
+            (
+                u'[u\'failed validation check for collection_model '
+                'field check of CollectionCommitLogEntryModel\', '
+                '[u"Model id collection-0-1: based on field collection_model '
+                'having value 0, expect model CollectionModel with id 0 '
+                'but it doesn\'t exist", u"Model id collection-0-2: based '
+                'on field collection_model having value 0, expect model '
+                'CollectionModel with id 0 but it doesn\'t exist"]]'
+            ), u'[u\'fully-validated CollectionCommitLogEntryModel\', 2]']
+        run_job_and_check_output(
+            self, expected_output, literal_eval=True)
+
+    def test_invalid_collection_version_in_model_id(self):
+        model_with_invalid_version_in_id = (
+            collection_models.CollectionCommitLogEntryModel.create(
+                '0', 3, self.owner_id, self.OWNER_USERNAME, 'edit',
+                'msg', [{}],
+                constants.ACTIVITY_STATUS_PUBLIC, False))
+        model_with_invalid_version_in_id.collection_id = '0'
+        model_with_invalid_version_in_id.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for collection model '
+                'version check of CollectionCommitLogEntryModel\', '
+                '[u\'Model id %s: Collection model corresponding '
+                'to collection id 0 has a version 1 which is less than '
+                'the version 3 in commit log model id\']]'
+            ) % (model_with_invalid_version_in_id.id),
+            u'[u\'fully-validated CollectionCommitLogEntryModel\', 3]']
+        run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_model_with_invalid_id(self):
+        model_with_invalid_id = (
+            collection_models.CollectionCommitLogEntryModel(
+                id='random-0-1', user_id=self.owner_id,
+                username=self.OWNER_USERNAME, commit_type='edit',
+                commit_message='msg', commit_cmds=[{}],
+                post_commit_status=constants.ACTIVITY_STATUS_PUBLIC,
+                post_commit_is_private=False))
+        model_with_invalid_id.collection_id = '0'
+        model_with_invalid_id.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for model id check of '
+                'CollectionCommitLogEntryModel\', '
+                '[u\'Model
```

Comment:
Use constants instead of public/private.

Reply:
Done.